surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/logger/SurefireProviderLoggerFactory.java

```diff
@@ -0,0 +1,63 @@
+package org.arquillian.smart.testing.surefire.provider.logger;
+
+import org.apache.maven.surefire.report.ConsoleLogger;
+import org.arquillian.smart.testing.logger.Logger;
+import org.arquillian.smart.testing.logger.LoggerFactory;
+
+import static java.lang.String.format;
+
+public class SurefireProviderLoggerFactory implements LoggerFactory {
+
+    private ConsoleLogger logger;
+    private boolean debugLogEnabled;
+
+    public SurefireProviderLoggerFactory(ConsoleLogger consoleLogger, boolean debugLogEnabled) {
+        this.logger = consoleLogger;
+        this.debugLogEnabled = debugLogEnabled;
+    }
+
+    @Override
+    public Logger getLogger() {
+        return new SurefireProviderLogger(logger);
+    }
+
+    private class SurefireProviderLogger implements Logger {
+
+        private static final String PREFIX = "%s: Smart Testing Extension - ";
```

Comment:
How about logging in the provider phase with the prefix `%s: Smart Testing Provider - `? Wouldn't that make more sense to have in maven extension phase: `[Smart Testing Extension]` and in provider with the relevant string? 
Or does it look just weird and confusing? WDYT @lordofthejars @bartoszmajsak 
And yeah, the default logger would contain only  `%s: Smart Testing - ` prefix. I'm also thinking in other contexts - Che or other IDEs

Reply:
Didn't we set it to `[]` thingy before for everything? Why is it now changed?

Reply:
We should have it unified for all so it's easy to grep etc.

Reply:
Yeah, you set it to `[ ]` because we wanted to make it unified with maven extension/plugin logging. Now, we have the possibility to have different prefixes in different stages of the maven build. 
When you take a look at the logging produced by surefire logic/providers, there is no `[ ]` present in the log. 
We should keep it aligned also with the logging of the particular stage.

Reply:
I am not really sure if that is so important to have it aligned with how surefire logs stuff, but maybe it makes sense from the point of view of seeing which side of the build it is - the build engine / extension vs plugin. If so, then let's make it this way.

Reply:
If you want to keep it as unified as possible, then I would remove brackets from the prefix at all. I mean from maven extension part `[Smart Testing Extension]`. When you check the maven log, there is not such a line containing multiple strings with brackets in sequence.
so in the maven extension, the log would start with:
`[INFO] Smart Testing Extension - `
and for provider side:
`INFO: Smart Testing Extension - `

Reply:
> I am not really sure if that is so important to have it aligned with how surefire logs stuff

because you don't log from maven extension but from the surefire side. For my eyes, it is better having one way of logging in particular phases - maven extension(or plugin) logs vs surefire logs...

Reply:
as for the content of the prefix, yeah I agree that it would be good having it unified - to grep it easily.

Reply:
I think we are on the same page here, no need to quote the first part of my previous comment, where I agree with unification based on the follow-up reasoning which you just rephrased.

Reply:
So, do we agree on: 
* `[INFO] Smart Testing Extension -` for maven extension logs
* `INFO: Smart Testing Extension -` for both the default logger and surefire provider logger

Reply:
Took us few extra comments, but YES we do :D

![agree](https://user-images.githubusercontent.com/719616/30694086-c9085eac-9ed2-11e7-9c03-7c3c964e4849.gif)

Reply:
I love your gifs :-D